_drafts/gdb-python-post.md

```diff
@@ -0,0 +1,378 @@
+$7 = (struct UuidListNode *) 0x20007088
+(gdb) p s_list_head->next->next->next
+$8 = (struct UuidListNode *) 0x20007070
+(gdb) p s_list_head->next->next->next->next
+$5 = (struct ListNode *) 0x0
+```
+
+While all this manual typing may be fine for debugging one crash or issue, it does not scale. When a new firmware is being QA'd for the first time on hundreds of devices internally, it's important to be able to quickly classify if the issue is known or not so debug doesn't cause the schedule to slip. Even once a product makes it into the wild, an issue that appears on .1% of devices in a one million device fleet would generate 1000 reports per month!
+
+Manually inspecting hundreds of crashes is no fun ... but fortunately, GDB has several facilities built in that allow you to [extend](https://sourceware.org/gdb/onlinedocs/gdb/Extending-GDB.html#Extending-GDB) its core functionality, including Python, the scripting language of choice for many embedded projects. This allows one to accelerate and in many cases even **automate** debug!
+
+In this series of posts we will explore some of the GDB Python APIs and how they can be used.
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest
+posts straight to your mailbox_
+
+## Getting started with GDB Python
+
+GDB's [Python API](https://sourceware.org/gdb/onlinedocs/gdb/Python.html#Python) was introduced as part of GDB 7.0 (all the way back in 2009!) and continues to receive new features and improvements as gdb is updated (look for "Python" in the [gdb release notes](https://www.gnu.org/software/gdb/news/))
+
+### Check if GDB was compiled with python support
+
+The GDB Python API is a GDB compile time option that can be enabled (with the `--with-python` configuration argument). We can easily check if gdb has this feature enabled by checking:
+
+```bash
+$ /path/to/gdb  --config
+This GDB was configured as follows:
+[...]
+             --with-python=/usr
+[...]
+```
+
+If you are doing any type of ARM Cortex-M development, you are in luck, the [official ARM Embedded Toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads) publishes a GDB variant with Python enabled at `./bin/arm-none-eabi-gdb-py` in each release!
+
+To use it you just need python installed on your system (i.e On ubuntu, `apt-get install python2.7 python-dev` or on OSX at the moment, the pre-installed system python is used)
+
+## Example Usage
+
+Let's walk through a trivial example snippet of code to get an idea of how some of the APIs work!
+
+The snippet tracks UUIDs in RAM in a linked list. On a real embedded system, we may have a bunch of items, identified by their UUID, stored on an external flash. If the list of items are looked at frequently, it may make sense for the list may be cached in RAM for faster lookup (since flash access can be slow) and better power (since reading from external flash consumes more power than a RAM read).
```

Comment:
Another weird sentence:
> if the list of items **are** looked at frequently, it **may make sense for the list** may be cached in RAM for faster lookup [...]

Should be
> If the list of items **is** looked at frequently, it may be cached in RAM for faster access and better power.